src/pages/resultsView/oncoprint/AddTracks.tsx

```diff
@@ -458,6 +465,75 @@ export default class AddTracks extends React.Component<IAddTrackProps, {}> {
         return Math.max(width, MIN_DROPDOWN_WIDTH);
     }
 
+    @action.bound
+    private addAll(clinicalAttributeIds: string[]) {
+        this.props.handlers.onChangeSelectedClinicalTracks!(
+            _.union(
+                this.getSelectedClinicalAttributes(),
+                clinicalAttributeIds.map(id => new ClinicalTrackConfig(id))
+            )
+        );
+    }
+
+    @action.bound
+    private clear(clinicalAttributeIds: string[]) {
+        this.props.handlers.onChangeSelectedClinicalTracks!(
+            _.differenceBy(
+                this.getSelectedClinicalAttributes(),
+                clinicalAttributeIds.map(id => new ClinicalTrackConfig(id)),
+                'stableId'
+            )
+        );
+    }
+
+    @action.bound
+    private toggleClinicalTrack(clinicalAttributeId: string) {
+        const toggled = toggleIncluded(
+            new ClinicalTrackConfig(clinicalAttributeId),
+            this.getSelectedClinicalAttributes(),
+            track => track.stableId === clinicalAttributeId
+        );
+        this.props.handlers.onChangeSelectedClinicalTracks!(toggled);
+    }
+
+    readonly addGroupTracksMenu = MakeMobxView({
+        await: () => [this.options],
+        render: () => (
+            <AddChartByType
+                options={this.options.result!.groups}
+                freqPromise={this.clinicalAttributeIdToAvailableFrequency}
+                onAddAll={this.addAll}
+                onClearAll={this.clear}
+                onToggleOption={this.toggleClinicalTrack}
+                optionsGivenInSortedOrder={true}
+                frequencyHeaderTooltip="% samples in group"
+                width={this.dropdownWidth}
+            />
+        ),
+        renderPending: () => <LoadingIndicator isLoading={true} small={true} />,
+        showLastRenderWhenPending: true,
+    });
+
+    readonly options = remoteData({
```

Comment:
I think this name could be improved (i realize it was existing).  It's really track options by type.    "trackOptionsByType" ?